youtube_dl/extractor/generic.py

```diff
@@ -3139,6 +3155,11 @@ def filter_video(urls):
             if embed_url and embed_url != url:
                 return self.url_result(embed_url)
 
+        if not found:
+            # DetikFlow: It's basically a 'modified' FlowPlayer used in https://20.detik.com
+            found = re.findall(
+                r'["\']videoUrl["\']\s*:\s*["\']([^"\']+)["\']', webpage)
```

Comment:
Generic extractor must have only generic embed patterns.

Reply:
Yes, it extracts any website containing `https://20.detik.com/embed/...` embed code.

Reply:
No. 2841-2847 does that. This code handles `20.detik.com/embed` page itself and it's not generic and should not be here.

Reply:
This is what happens without line 3158-3161:
```
python -m youtube_dl -v -F https://20.detik.com/e-flash/20180330-180330036/ready-yuk-party-bareng-the-chainsmokers-malam-ini
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-F', 'https://20.detik.com/e-flash/20180330-180330036/ready-yuk-party-bareng-the-chainsmokers-malam-ini']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] youtube-dl version 2018.03.26.1
[debug] Git HEAD: 14f4c41ce
[debug] Python version 3.6.4 (CPython) - Windows-10-10.0.16299-SP0
[debug] exe versions: ffmpeg N-89951-g3caecf7ce8, ffprobe N-89951-g3caecf7ce8
[debug] Proxy map: {}
[generic] ready-yuk-party-bareng-the-chainsmokers-malam-ini: Requesting header
WARNING: Falling back on generic information extractor.
[generic] ready-yuk-party-bareng-the-chainsmokers-malam-ini: Downloading webpage
[generic] ready-yuk-party-bareng-the-chainsmokers-malam-ini: Extracting information
[generic] 180330036?autostart=1: Requesting header
WARNING: Falling back on generic information extractor.
[generic] 180330036?autostart=1: Downloading webpage
[generic] 180330036?autostart=1: Extracting information
ERROR: Unsupported URL: https://20.detik.com/embed/180330036?autostart=1
Traceback (most recent call last):
  File "E:\Project (2018)\youtube-dl\youtube_dl\YoutubeDL.py", line 785, in extract_info
    ie_result = ie.extract(url)
  File "E:\Project (2018)\youtube-dl\youtube_dl\extractor\common.py", line 440, in extract
    ie_result = self._real_extract(url)
  File "E:\Project (2018)\youtube-dl\youtube_dl\extractor\generic.py", line 3164, in _real_extract
    raise UnsupportedError(url)
youtube_dl.utils.UnsupportedError: Unsupported URL: https://20.detik.com/embed/180330036?autostart=1
```

Reply:
Do you read my messages at all? I did not say to remove it completely. It must be moved to a separate extractor.

Reply:
Okay, got it. Thank you.